x-pack/plugins/enterprise_search/server/lib/indices/pipelines/ml_inference/update_ml_inference_mappings.ts

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ElasticsearchClient } from '@kbn/core-elasticsearch-server';
+
+import { FieldMapping } from '../../../../../common/ml_inference_pipeline';
+
+import { ErrorCode } from '../../../../../common/types/error_codes';
+
+import { isIllegalArgumentException } from '../../../../utils/identify_exceptions';
+
+/**
+ * Creates Elasticsearch field mappings so that the outputs of ML Inference pipelines are not indexed as `float` fields
+ * @param indexName - the index whose mapping will be updated
+ * @param fieldMappings - the array of objects representing the source field (text) names and target fields (ML output) names
+ * @param esClient
+ */
+export const updateMlInferenceMappings = async (
+  indexName: string,
+  fieldMappings: FieldMapping[],
+  esClient: ElasticsearchClient
+) => {
+  const sourceFields = fieldMappings.map((fieldMapping) => {
+    return fieldMapping.sourceField;
+  });
+
+  const nonDefaultTargetFields = fieldMappings
+    .filter((fieldMapping) => {
+      return fieldMapping.targetField !== `ml.inference.${fieldMapping.sourceField}_expanded`;
```

Comment:
I'm not sure about this filter. We do want to put `targetField` mappings for every selected `sourceField`, and for now it's always `ml.inference.${fieldMapping.sourceField}_expanded`.
Wouldn't this filter exclude all target fields from the mapping update statement?

Reply:
Yes. If you keep reading, the `nonDefaultTargetFields` end up being `[]` for now, but the right mappings are getting generated from the `sourceFields`.

This is because I'd originally designed this expecting that the user could specify custom targets in 8.8. Even though they can't, eventually they will be able to, and this code will be ready for that.